plugin_interpreter/src/controller_plugin.py

```diff
@@ -83,11 +91,23 @@ def __init__(self, name, functionality):
         function usage. It will be displayed to the user through
         the interface.
         """
-        self.name = name
         """Define server port/proto requirement (TCP/UDP) so docker can be run
         properly."""
         super().__init__()
 
+    def _read_functionality(self):
+        curr_dir = ospath.dirname(ospath.dirname(__file__))
+        # filename = curr_dir + "/plugins/__" + self.name + "/" + self.name + ".json"
+        filename = "{}/plugins/__{name}/{name}.json".format(curr_dir, name = self.name)
+        try:
+            with open(filename) as f:
+                self.functionality = json.load(f)
+        except (IOError, json.JSONDecodeError):
+            self.functionality = [{"name": "Functionality Error",}]
```

Comment:
can you make this base/error functionality match the requirements so we don't throw errors upstream?